### DISCUSSION TITLE
suppress tbl-cap?

### DISCUSSION THREAD

**Discussion by thm181** [quarto-dev/quarto-cli#4209] [Q&A]
Is there a way to suppress the YAML expectation of tbl-cap when using kable, gt, or other table package that has built-in caption capabilities? This would allow us to continue to use captions that include markdown and improve backwards compatibility. If I just don't include tbl-cap:, I get, "Table #: ?(caption)" then the caption I've provided, e.g. tab_caption(md("my table caption"), as a sub-caption.

**Comment by cscheid** [comment]
Can you provide a complete example where quarto is behaving differently from how you'd prefer it to?

We're currently revamping the support for table captions and sub-captions from computational output, so this general area will be improved in the near future.

**Reply by thm181** [reply]
For HTML output . . . 

````
```{r}
#| label: setup
#| echo: true
#| message: false

knitr::opts_chunk$set(
  echo = TRUE,
  message = TRUE
  ) 

library(gt)          
library(tidyverse)

```

```{r}
#| label: tbl-YAMLcaption
#| tbl-cap: "Attributes of three species of *Iris*."

data(iris)

gt(
  head(iris)
  )
```

```{r}
#| label: tbl-noYAMLcaption

data(iris)

gt(
  head(iris),
   caption = md("Attributes of three species of *Iris*.")
   )
```
````

This was an attempt to figure out a work-around for not being able to use markdown in the YAML caption option.

**Reply by thm181** [reply]
As a follow-up, fig-cap allows markdown, but tbl-cap doesn't.  It would also be *nice* if we could have table captions width adjustment capabilities. I often end up needing to add empty columns to a table to improve the readability of the caption.

**Reply by cscheid** [reply]
Just a quick update.

> As a follow-up, fig-cap allows markdown, but tbl-cap doesn't.

This behavior is going to change on 1.3, and in ways that I believe will better support your use case.

For example. If you use the latest versions (from github) of `gt` _and_ quarto, markdown rendering works _everywhere_ on your tables (not only captions, but formatted output, etc), and quarto will resolve references correctly (that came from https://github.com/rstudio/gt/pull/1186/files). @rich-iannone and I are coordinating work with other table library authors to make sure they also support rich Markdown input in the upcoming 1.3 release.

> It would also be nice if we could have table captions width adjustment capabilities

I'm not quite sure what you mean here, but if you can provide a minimal example, we can try to address it. Thanks!

**Reply by thm181** [reply]
That's great news! 

I, and I assume others, can easily work around the caption width issue, so it's far from needing immediate attention.  Currently the caption width is set by the table width, so if you do something like: kable_classic(full = FALSE) and there are only two small columns, the caption can get squeezed into something only a few characters wide.
